Hadrons/Modules/MContraction/Meson.hpp

```diff
@@ -188,8 +221,9 @@ void TMeson<FImpl1, FImpl2>::execute(void)
     Gamma                  g5(Gamma::Algebra::Gamma5);
     std::vector<GammaPair> gammaList;
     int                    nt = env().getDim(Tp);
-    
-    parseGammaString(gammaList);
+    envGetTmp(std::vector<int>, gammaSnk);
```

Comment:
You do not need to do a temporary with `gammaSnk`, the object is negligible and not allocated by Grid anyway.

Reply:
Thanks Antonin. I agree with what you ask for. The module relied on the integer representation of the gamma matrices beforehand as well - I guess that part should be changed also then? When choosing "all" as an input it would loop over the odd integers from 1 to 31 to select the correct ones.

Reply:
I don't have time this evening but maybe will sometime this week. But like you say, we might start first runs with this version

Reply:
Yes the original was also not ideal, but it was confined to the `"all"` generation, now this is spreading more. It also pre-dated some of the re-organisation of gamma matrices in Grid (it was one of the first module ever!). You can actually use `Gamma::gall` which is a static constant array containing the 16 matrices if you want to do a loop on all of them without refering to their integer values. See https://github.com/paboyle/Grid/blob/develop/Grid/qcd/spin/Gamma.h.

Since we are refactoring that quite a bit getting that in a more elegant way will probably be beneficial for the future. But as said we can see that later and you should start running if there is no time.